test/Elastic.Apm.MongoDb.Tests/MongoApmTests.cs

```diff
@@ -0,0 +1,159 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Elastic.Apm.Api;
+using Elastic.Apm.Config;
+using Elastic.Apm.MongoDb.Tests.Fixture;
+using Elastic.Apm.Tests.Utilities;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using Moq;
+using Xunit;
+
+namespace Elastic.Apm.MongoDb.Tests
+{
+	public class MongoApmTests : IClassFixture<MongoFixture<MongoApmTests.MongoConfiguration, BsonDocument>>,
+		IDisposable
+	{
+		public MongoApmTests(MongoFixture<MongoConfiguration, BsonDocument> fixture)
+		{
+			_documents = fixture.Collection ?? throw new ArgumentNullException(nameof(fixture.Collection));
+			_payloadSender = new MockPayloadSender();
+
+			var configurationReaderMock = new Mock<IConfigurationReader>();
+			configurationReaderMock.Setup(x => x.TransactionSampleRate)
+				.Returns(() => 1.0);
+			configurationReaderMock.Setup(x => x.TransactionMaxSpans)
+				.Returns(() => 50);
+			configurationReaderMock.Setup(x => x.Enabled)
+				.Returns(() => true);
+			configurationReaderMock.Setup(x => x.Recording)
+				.Returns(() => true);
+
+			var config = new AgentComponents(configurationReader: configurationReaderMock.Object,
```

Comment:
Can be replaced with `TestAgentComponents` and a `MockConfigSnapshot`

Reply:
👍  done.